libmeepgeom/meepgeom.cpp

```diff
@@ -1209,50 +1133,68 @@ static double get_chi2(meep::component c, const medium_struct *m) {
   }
 }
 
-bool geom_epsilon::has_chi2(meep::component c)
+static double get_chi(meep::component c, const medium_struct *m, int p)
+ { return ((p==2) ? get_chi2(c,m) : get_chi3(c,m)); }
+
+// the following routines consolidate the former has_chi2, has_chi3 and chi2, chi3
+// p=2,3 for chi2,chi3
+bool geom_epsilon::has_chi(meep::component c, int p)
 {
   medium_struct *mm;
 
   for (int i = 0; i < geometry.num_items; ++i)
-   if (    is_medium(geometry.items[i].material, &mm)
-        && get_chi2(c, mm) != 0
-      ) return true;
+   if ( is_medium_or_user(geometry.items[i].material, &mm) )
+    if (get_chi(c, mm, p)!=0)
```

Comment:
Would be clearer to just remove the `!= 0` since `get_chi` returns a boolean (0/1) value.

Reply:
`get_chi` returns a `double` value. (Almost all of the code here is taken directly from the original; I just eliminated some redundancy between the chi2/chi3 cases.)